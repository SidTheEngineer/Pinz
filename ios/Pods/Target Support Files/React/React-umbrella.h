#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

