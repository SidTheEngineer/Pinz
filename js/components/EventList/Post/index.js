import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  Text,
  View,
  Platform
} from 'react-native';

import UpvoteDownvote from './UpvoteDownvote';
import { COLORS } from '../../../constant';

const styles = StyleSheet.create({
  postContainer: {
    height: 100,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
    marginBottom: 6,
    backgroundColor: 'white',
    marginLeft: 10,
    marginRight: 10,
    elevation: 1,
    shadowOffset: {
      height: 0.5,
      width: -0.5
    },
    shadowOpacity: 0.3,
    shadowColor: 'black',
    shadowRadius: 1
    // borderColor: 'rgb(150,150,150)',
    // borderStyle: 'solid',
    // borderTopWidth: 0.5,
    // borderBottomWidth: 0.5,
  },
  textContainer: {
    flex: 1,
    flexDirection: 'column',
    margin: 10
  },
  title: {
    flex: 0.2,
    fontWeight: 'bold'
  },
  text: {
    flex: 0.8,
    color: COLORS.GRAY,
    padding: 2,
    fontSize: Platform.OS === 'ios' ? 12 : 14,
  }
});

const Post = props => (
  <TouchableOpacity style={styles.postContainer}>
    <UpvoteDownvote />
    <View style={styles.textContainer}>
      <Text style={[styles.text, styles.title]}>
        {props.details.title.replace(/<(?:.|\n)*?>/gm, '')}
      </Text>
      <Text style={styles.text} ellipsizeMode="tail">
        {/* Replace HTML tags that are in UCF's event descriptions. */}
        {/* Strip blank lines in description. */}
        {
          props.details.description.length > 200
            ? `${
                  props.details.description
                    .replace(/<(?:.|\n)*?>/gm, '')
                    .replace(/^\s*[\r\n]/gm, '')
                    .trim()
                    .substring(0, 200)
                }...`
            : props.details.description
        }
      </Text>
    </View>
  </TouchableOpacity>
);

Post.propTypes = {
  details: React.PropTypes.shape({
    title: React.PropTypes.string.isRequired,
    description: React.PropTypes.string.isRequired
  })
};

export default Post;
