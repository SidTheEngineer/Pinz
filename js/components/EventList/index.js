import React, { Component } from 'react';
import {
  StyleSheet,
  ListView,
  InteractionManager,
  View,
  ActivityIndicator
} from 'react-native';
import Post from './Post';

import { COLORS } from '../../constant';

const styles = StyleSheet.create({
  listContainer: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: COLORS.LIGHT_GRAY,
    marginTop: 60,
  },
  loadingContainer: {
    flex: 1,
    backgroundColor: COLORS.LIGHT_GRAY,
    alignItems: 'center',
    justifyContent: 'center'
  }
});

class EventList extends Component {

  static propTypes = {
    map: React.PropTypes.shape({
      events: React.PropTypes.oneOfType([
        React.PropTypes.arrayOf(React.PropTypes.object),
        React.PropTypes.object
      ])
    })
  }

  // While loading posts.
  static renderPlaceholderView() {
    return (
      <View style={styles.loadingContainer}>
        <ActivityIndicator
          style={{ marginBottom: 100 }}
          size={50}
          color={COLORS.LIGHT_GREEN}
          animating
        />
      </View>
    );
  }

  constructor(props) {
    super(props);

    const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
    this.state = {
      dataSource: ds.cloneWithRows(this.props.map.events.map(event => <Post details={event} />)),
      renderPlaceholderOnly: true
    };
  }

  componentDidMount() {
    InteractionManager.runAfterInteractions(() => {
      this.setState({ renderPlaceholderOnly: false });
    });
  }

  render() {
    if (this.state.renderPlaceholderOnly) {
      return EventList.renderPlaceholderView();
    }

    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={rowData => rowData}
        style={styles.listContainer}
        contentContainerStyle={{ justifyContent: 'space-between' }}

        // Ratio of initial render to how many rendered
        // per frame.
        initialListSize={10}
        pageSize={10}
        // scrollRenderAheadDistance={400}
      />
    );
  }
}

export default EventList;
