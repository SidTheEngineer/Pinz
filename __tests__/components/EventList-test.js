import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventList from '../../js/components/EventList';

const props = {
  map: {
    events: [
      {
        event_id: '12345',
        id: '12345',
        title: 'This is a test event',
        description: 'test test test',
        location: 'Location 123',
        starts: 'midnight',
        ends: 'midnight',
        category: 'Sports'
      },
      {
        event_id: '123456',
        id: '123456',
        title: 'This is a test event',
        description: 'test test test',
        location: 'Location 123',
        starts: 'midnight',
        ends: 'midnight',
        category: 'Sports'
      }
    ]
  }
};

describe('EventList', () => {
  it('renders correctly', () => {
    const eventList = renderer.create(
      <EventList {...props} />
    ).toJSON();
    expect(eventList).toMatchSnapshot();
  });

  it('renders an ActivityIndicator', () => {
    const placeholderView = renderer.create(
      EventList.renderPlaceholderView()
    ).toJSON();

    expect(placeholderView).toMatchSnapshot();
  });
});
