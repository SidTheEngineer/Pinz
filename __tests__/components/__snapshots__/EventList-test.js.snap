exports[`EventList renders an ActivityIndicator 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(235,235,235)",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <ActivityIndicator
    animating={true}
    color="rgb(51, 204, 94)"
    hidesWhenStopped={true}
    size={50}
    style={
      Object {
        "marginBottom": 100,
      }
    } />
</View>
`;

exports[`EventList renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(235,235,235)",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <ActivityIndicator
    animating={true}
    color="rgb(51, 204, 94)"
    hidesWhenStopped={true}
    size={50}
    style={
      Object {
        "marginBottom": 100,
      }
    } />
</View>
`;
